lexer/lexer.go

```diff
@@ -51,6 +51,16 @@ func (l *Lexer) NextToken() token.Token {
 		tok = newToken(token.COMMA, l.ch)
 	case ';':
 		tok = newToken(token.SEMICOLON, l.ch)
+	case '=':
+		tok = newToken(token.EQUALS, l.ch)
+	case '!':
+		if string(l.input[l.position:l.position+2]) == "!=" {
+			tok = token.Token{Type: token.NOTEQUALS, Literal: "!="}
+			l.readChar()
+			l.readChar()
+			return tok
+		}
```

Comment:
This feels suboptimal. Essentially I want to peek at the next character, but the lexer only looks at one character at a time, sort of only goes forward. But for the character `!`, it's not clear until we see the next character whether we want the bang (prefix) operator, or the `!=` (infix) operator